docs/decisions/20240103-use-jwt-claims-for-agent-admin-auth.md

```diff
@@ -0,0 +1,155 @@
+# Use JWT claims for agent admin access control
+
+- Status: accepted
+- Deciders: Pat Losoponkul, Yurii Shynbuiev, David Poltorak, Shailesh Patil
+- Date: 2024-01-03
+- Tags: multitenancy, authorisation, authentication
+
+Technical Story: [Allow agent admin role to be authenticated by Keycloak JWT | https://input-output.atlassian.net/browse/ATL-6074]
+
+## Context and Problem Statement
+
+External IAM authentication and authorization are currently limited to tenants using the wallets.
+Administrators currently rely on a static API key configured in the agent.
+Enhancing the security of admin access requires extending support for admin authentication through an external IAM.
+This introduces a concept of role to the authentication and authorization process.
```

Comment:
Role-based access control (RBAC) or Attribute-based Access Control (ABAC) are general concepts that we can implement to solve the issue ^^^.
So, `This introduces,` from my point of view, should be removed from the Problem statement.

I would like to make a problem statement short and clear: centralized permission management of the access of the Agent's Admin API, standard and security-compliant approach, simplicity, decoupling, and extendability of the solution.

Reply:
fixed